provider/spark_configs.go

```diff
@@ -349,6 +429,119 @@ func (flag sparkDeployFlag) SparkFlags() sparkFlags {
 	}
 }
 
+type sparkSnowflakeFlags struct {
+	Config       *pc.SnowflakeConfig
+	ExecutorType pc.SparkExecutorType
+}
+
+func (args sparkSnowflakeFlags) SparkFlags() sparkFlags {
+	if args.Config == nil {
+		logging.GlobalLogger.Debug(
+			"Not setting spark snowflake flags, snowflake config not set",
+		)
+		return sparkFlags{}
+	}
+	flags := sparkFlags{
+		sparkCredFlag{
+			Key:   "sfURL",
+			Value: args.Config.GetBaseURL(),
+		},
+		sparkCredFlag{
+			Key:   "sfUser",
+			Value: args.Config.Username,
+		},
+		sparkCredFlag{
+			Key:   "sfPassword",
+			Value: args.Config.Password,
+		},
+		sparkCredFlag{
+			Key:   "sfWarehouse",
+			Value: args.Config.Warehouse,
+		},
+	}
+	// Databricks has these packages pre-installed.
+	if args.ExecutorType == pc.EMR {
+		flags = append(flags, sparkPackagesFlag{
+			Packages: []string{
+				"net.snowflake:snowflake-jdbc:3.13.22",
+				"net.snowflake:spark-snowflake_2.12:2.12.0-spark_3.4",
```

Comment:
I'm assuming the versions of these packages could change so maybe we want to put this something that can allow for updates without a new release, yeah?